### PR TITLE
Reorganize/edit/update some sections

### PR DESCRIPTION
**Pull request by harendra-kumar**
* Remove CI badges, these are on the wiki page now.
* Move the internal modules section at the end
* Remove redundant links about docs, and move it down after install/use

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,31 +15,27 @@ software systems.  The key features it provides are:
 * Powerful abstractions
 * Declarative concurrency
 
-Let's go through some practical examples to see it working. You
-can find the working code of these examples in the [streamly-examples
+This guide introduces you to streamly using some
+practical examples to see it working. You can find the
```
Comment (adithyaov) [inline README.md:19]:
```suggestion
practical examples. You can find the
```

File: README.md [hunk 2]
```diff
@@ -27,31 +15,27 @@ software systems.  The key features it provides are:
...
+working code of these examples in the [streamly-examples
```
Comment (adithyaov) [inline README.md:20]: Will this link be rendered properly?

File: README.md (hunk 1 above)
Comment (harendra-kumar) [reply README.md:19]: ok.

File: README.md (hunk 2 above)
Comment (harendra-kumar) [reply README.md:20]: Yes. See https://github.com/composewell/streamly/tree/readme .

File: README.md [hunk 3]
```diff
@@ -93,6 +77,9 @@ wcb file =
   & Stream.fold Fold.length  -- IO Int
 ```
 
+If you are wondering about the `Internal` module import, see [A Note on
+Internal Modules](#a-not-on-internal-modules) section.
```
Comment (adithyaov) [inline README.md]: Broken link. Should be `#a-note-on-internal-modules`.

```suggestion
Please go through [this](#a-note-on-internal-modules) section 
for more details on internal modules.
```

Comment (harendra-kumar) [reply README.md]: fixed already!

File: README.md [hunk 4]
```diff
@@ -529,13 +515,25 @@ We hope to bring these optimizations to GHC in future but until
...
-## Support
+## A Note on Internal Modules
 
-Please feel free to ask questions on the
-[streamly gitter channel](https://gitter.im/composewell/streamly).
-If you require professional support, consulting, training or timely
-enhancements to the library please contact
-[support@composewell.com](mailto:support@composewell.com).
+You may see some `Internal` modules imported in the examples. Those are
+not really Internal to the library.  We can classify streamly modules in
```
Comment (adithyaov) [inline README.md:521]:
```suggestion
not really Internal to the library.  We can classify streamly modules into
```

File: README.md [hunk 5]
```diff
@@ -529,13 +515,25 @@ We hope to bring these optimizations to GHC in future but until
...
+two categories:
+
+* Released Modules: these are modules that have a stable API, any API changes
```
Comment (adithyaov) [inline README.md:524]:
```suggestion
* Released Modules: These are modules that have a stable API, any API changes
```

File: README.md [hunk 6]
```diff
@@ -529,13 +515,25 @@ We hope to bring these optimizations to GHC in future but until
...
+  conform to a versioning policy.
```
Comment (adithyaov) [inline README.md:525]:
```suggestion
  conform to the versioning policy.
```

File: README.md [hunk 7]
```diff
@@ -529,13 +515,25 @@ We hope to bring these optimizations to GHC in future but until
...
+* Pre-release Modules: These modules are not yet released due to some planned
+  changes in near future, they will be released soon. These modules reside in
+  the `Internal` namespace for now.
```
Comment (adithyaov) [inline README.md:528]: I'm unable to phrase this properly. But replace `some planned changes in near future` with something regarding having a soak time for the API.

File: README.md [hunk 8]
```diff
@@ -550,8 +548,18 @@ See the `credits` directory for full list of contributors, credits and licenses.
...
+Streamly is an [open source](https://github.com/composewell/streamly)
+project available under [BSD-3-Clause
+license](https://github.com/composewell/streamly/blob/master/LICENSE).
+
+## Support
+
+* Join streamly [online community chat](https://gitter.im/composewell/streamly)
+* For professional support please contact [support@composewell.com](mailto:support@composewell.com)
+
+# [Streamly Homepage](https://streamly.composewell.com)
```
Comment (adithyaov) [inline README.md:565]: This can be removed.

**Review by adithyaov — APPROVED** [review]
Looks good. Minor comments.

File: README.md (hunk 4 above)
Comment (harendra-kumar) [reply README.md:521]: ok.

File: README.md (hunk 5 above)
Comment (harendra-kumar) [reply README.md:524]: ok.

File: README.md (hunk 6 above)
Comment (harendra-kumar) [reply README.md:525]: "the package versioning policy".

File: README.md (hunk 8 above)
Comment (harendra-kumar) [reply README.md:565]: Need to have the homepage link at some prominent place. I add a reference in the intro para then.

**Pull request merged**